framework/src/engine/engine.ts

```diff
@@ -203,6 +204,12 @@ export class Engine {
 			logger: this._logger,
 			chainID: this._chain.chainID,
 		});
+
+		const legacyEndpoint = new LegacyEndpoint({
+			db: this._legacyDB,
+		});
+		legacyEndpoint.init(this._blockchainDB);
```

Comment:
I think `legacyEndpoint` won't need blockchain.db. maybe it was my mistake to add `init` function on bootstrap, so feel free to remove this